sphinx/ext/intersphinx.py

```diff
@@ -273,22 +275,43 @@ def load_mappings(app: Sphinx) -> None:
         # Duplicate values in different inventories will shadow each
         # other; which one will override which can vary between builds
         # since they are specified using an unordered dict.  To make
-        # it more consistent, we sort the named inventories and then
-        # add the unnamed inventories last.  This means that the
-        # unnamed inventories will shadow the named ones but the named
-        # ones can still be accessed when the name is specified.
-        named_vals = []
-        unnamed_vals = []
+        # it more consistent, we sort the named inventories by their
+        # name and then add the unnamed inventories last.  This means
+        # that the unnamed inventories will shadow the named ones but
+        # the named ones can still be accessed when the name is specified.
+        #
+        # When we encounter a named inventory that already exists,
+        # this means that we had two entries for the same inventory,
+        # but with different URIs (e.g., the remote documentation was
+        # updated).  In particular, the newest URI is inserted in the
+        # cache *after* the one that existed from a previous build.
+        #
+        # Example: assume that we use A to generate the inventory of "foo",
+        # so that we have
+        #
+        #   intersphinx_cache = {A: ('foo', timeout, D1)}
+        #
+        # If we rebuild the project, the cache becomes
+        #
+        #   intersphinx_cache = {A: ('foo', ..., D1), B: ('foo', ..., D2)}
+        #
+        # So if we iterate the values of ``intersphinx_cache``, we correctly
+        # replace old inventory cache data with the same name but different
+        # URIs. If we have the same URI or if we force-reload the cache, the
+        # data is already updated.
```

Comment:
What's the lifetime of the `intersphinx_cache` in a typical deployment, and how is it stored?

Reply:
(and I guess to explain more of my understanding: there's one `objects.inv` file for each remote documentation site in use with intersphinx)

Reply:
The objects.inv is actually stored in the build so it is served as a static page. It's just that, by default, it's served at the root.

Reply:
Hrm, OK.  Updating my understanding: each build with intersphinx enabled may pull in references from multiple external projects, and the results are written to a single local `objects.inv` file.

And the `intersphinx_cache` (a configurable cache) is anything that implements the [`intersphinx_cache: dict[str, InventoryCacheEntry]`](https://github.com/sphinx-doc/sphinx/blob/fb1e5212021b02a08e7bfcf6395107dfe3cb3108/sphinx/ext/intersphinx.py#L257) interface?  (with a [fallback of a basic Python `dict` if nothing is configured](https://github.com/sphinx-doc/sphinx/blob/fb1e5212021b02a08e7bfcf6395107dfe3cb3108/sphinx/ext/intersphinx.py#L69-L71)).

Reply:
(my confusion here is largely because I don't understand why a `intersphinx_cache_limit` would exist if the cache itself is backed by an in-process `dict` instance)

Reply:
The `intersphinx_cache_limit` is set to avoid refetching the content of the URL everytime.

In general, you keep the same URL and doesn't change it once. However, if you do that, maybe there are new releases inbetween and your inventory is outdated. So you redownload objects.inv to update your cache entry.